clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -467,37 +467,48 @@ else if (strategy == OffsetResetStrategy.LATEST)
      *         the defaultResetPolicy is NONE
      */
     public Map<TopicPartition, List<ConsumerRecord<K, V>>> fetchedRecords() {
+        if (nextInLineException != null) {
```

Comment:
What if the subscription changes and the partition which had exception is not assigned anymore? Or if user seeks to a new position for the partition in question which should no longer throw exception? It seems we need to verify whether the partition state has changed or not before throw the cached exception.